src/main/java/tc/oc/pgm/timelimit/TimeLimitModule.java

```diff
@@ -1,103 +1,89 @@
 package tc.oc.pgm.timelimit;
 
-import java.util.Set;
 import java.util.logging.Logger;
 import javax.annotation.Nullable;
 import org.jdom2.Attribute;
 import org.jdom2.Document;
 import org.jdom2.Element;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.MapProtos;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
 import tc.oc.pgm.api.match.Match;
-import tc.oc.pgm.bossbar.BossBarModule;
-import tc.oc.pgm.map.MapModule;
-import tc.oc.pgm.map.MapModuleContext;
-import tc.oc.pgm.map.ProtoVersions;
-import tc.oc.pgm.maptag.MapTag;
-import tc.oc.pgm.module.ModuleDescription;
+import tc.oc.pgm.api.match.MatchModule;
 import tc.oc.pgm.result.VictoryCondition;
 import tc.oc.pgm.result.VictoryConditions;
 import tc.oc.pgm.util.XMLUtils;
 import tc.oc.util.components.PeriodFormats;
 import tc.oc.xml.InvalidXMLException;
 
-@ModuleDescription(
-    name = "Time Limit",
-    requires = {BossBarModule.class})
-public class TimeLimitModule extends MapModule<TimeLimitMatchModule> {
-
-  private static final MapTag TIMELIMIT_TAG = MapTag.forName("timelimit");
-
+public class TimeLimitModule implements MapModule {
   private final @Nullable TimeLimit timeLimit;
 
   public TimeLimitModule(@Nullable TimeLimit limit) {
     this.timeLimit = limit;
   }
 
   @Override
-  public void loadTags(Set<MapTag> tags) {
-    if (timeLimit != null) tags.add(TIMELIMIT_TAG);
-  }
```

Comment:
Why complete removal?

Reply:
Too many tags.. you have to keep it simple.

Reply:
You could easily differentiate maps where you have to score n amount of points to win the match and maps where you have to score most points before the time ends. This one was pretty useful.